### DISCUSSION TITLE
Purpose of the `spec` directory?  How to build a stanalone`tl` binary?

### DISCUSSION THREAD

**Discussion by parke** [teal-language/tl#471] [Q&A] [answered]
What is the purpose of the files in the `spec` directory?

https://github.com/teal-language/tl/tree/master/spec

The Linux release tarball contains a standalone binary executable version of `tl`.

https://github.com/teal-language/tl/releases/tag/v0.13.2

How do I build a copy of this standalone binary executable from source?

**Comment by hishamhm** [comment, marked as answer]
> What is the purpose of the files in the spec directory?

These are tests, which you can run with `luarocks test`.

> How do I build a copy of this standalone binary executable from source?

This particular binary shipped is built using a custom script which I have not published. But the `tl` compiler is a pure-Lua application, which depends on other modules (argparse and luafilesystem). There are various tools that can amalgamate multiple Lua sources into a single executable; [https://github.com/jirutka/luapak/](luapak), for example, is one of them, and it should be able to produce equivalent results.
